### PR TITLE
Implement "List Active Guild Threads"

### PR DESCRIPTION
**Pull request by myrkvi**
Creates a method that implements the [`/guilds/{guild.id}/threads/active`][ref] endpoint of the Discord API.

[ref]: https://discord.com/developers/docs/resources/guild#list-active-guild-threads

### CODE REVIEW

File: rest/rest_endpoints.go [hunk 1]
```diff
@@ -60,6 +60,7 @@ var (
 	CreateGuildChannel     = NewEndpoint(http.MethodPost, "/guilds/{guild.id}/channels")
 	GetGuildChannels       = NewEndpoint(http.MethodGet, "/guilds/{guild.id}/channels")
 	UpdateChannelPositions = NewEndpoint(http.MethodPatch, "/guilds/{guild.id}/channels")
+	GetActiveGuildThreads  = NewEndpoint(http.MethodGet, "/guilds/{guild.id}/threads/active")
```
Comment (sebm253) [inline rest/rest_endpoints.go:63]: can you please move this into the part with Thread endpoints? L181

File: rest/guilds.go [hunk 2]
```diff
@@ -123,6 +124,11 @@ func (s *guildImpl) UpdateChannelPositions(guildID snowflake.ID, guildChannelPos
 	return s.client.Do(UpdateChannelPositions.Compile(nil, guildID), guildChannelPositionUpdates, nil, opts...)
 }
 
+func (s *guildImpl) GetActiveGuildThreads(guildID snowflake.ID, opts ...RequestOpt) (activeThreads discord.GuildActiveThreads, err error) {
```
Comment (sebm253) [inline rest/guilds.go:127]: this should return a pointer of the struct

File: rest/guilds.go [hunk 3]
```diff
@@ -27,6 +27,7 @@ type Guilds interface {
 	CreateGuildChannel(guildID snowflake.ID, guildChannelCreate discord.GuildChannelCreate, opts ...RequestOpt) (discord.GuildChannel, error)
 	GetGuildChannels(guildID snowflake.ID, opts ...RequestOpt) ([]discord.GuildChannel, error)
 	UpdateChannelPositions(guildID snowflake.ID, guildChannelPositionUpdates []discord.GuildChannelPositionUpdate, opts ...RequestOpt) error
+	GetActiveGuildThreads(guildID snowflake.ID, opts ...RequestOpt) (discord.GuildActiveThreads, error)
```
Comment (sebm253) [inline rest/guilds.go:30]: same here - pointer

File: rest/rest_endpoints.go (hunk 1 above)
Comment (myrkvi) [reply rest/rest_endpoints.go:63]: Sure! Just to confirm, this is the case even though it is part of the Guilds endpoints on Discord's end?

Comment (sebm253) [reply rest/rest_endpoints.go:63]: correct - we separate them like this even though they share a common endpoint

File: rest/guilds.go [hunk 4]
```diff
@@ -27,6 +27,7 @@ type Guilds interface {
 	CreateGuildChannel(guildID snowflake.ID, guildChannelCreate discord.GuildChannelCreate, opts ...RequestOpt) (discord.GuildChannel, error)
 	GetGuildChannels(guildID snowflake.ID, opts ...RequestOpt) ([]discord.GuildChannel, error)
 	UpdateChannelPositions(guildID snowflake.ID, guildChannelPositionUpdates []discord.GuildChannelPositionUpdate, opts ...RequestOpt) error
+	GetActiveGuildThreads(guildID snowflake.ID, opts ...RequestOpt) (*discord.GuildActiveThreads, error)
```
Comment (sebm253) [inline rest/guilds.go:30]: it would make sense to also move this func into rest/threads

**Review by sebm253 — CHANGES_REQUESTED** [review]

**Review by sebm253 — APPROVED** [review]
thanks :)

**Pull request merged**